### PR TITLE
add analytics documentation

### PR DESCRIPTION
**Pull request by wilk10**
[XN-1680](https://xainag.atlassian.net/browse/XN-1680)

Add documentation to Federated Analytics crate.

Please let me know if it's clear enough.

### CODE REVIEW

File: rust/xaynet-analytics/src/database/common.rs [hunk 1]
```diff
@@ -41,6 +50,11 @@ pub trait IsarAdapter<'object>: Sized {
     }
 }
 
+/// This trait is implemented directly for each data model to have a high level API for `AnalyticsController` to
+/// save/get obejcts from the db.
```
Comment (finiteprods) [inline rust/xaynet-analytics/src/database/common.rs:54]:
```suggestion
/// save/get objects from the db.
```

File: rust/xaynet-analytics/src/database/common.rs [hunk 2]
```diff
@@ -74,6 +89,11 @@ impl FieldProperty {
     }
 }
 
+/// `SchemaGenerator` is needed to register the `PropertySchema` and `IndexSchema` for each `FieldProperty`.
+/// `PropertySchema` and `IndexSchema` are imported from Isar, while `FieldProperty` is an internal struct to
+/// make it conventient to iterate through each property (see the fold below).
```
Comment (finiteprods) [inline rust/xaynet-analytics/src/database/common.rs:94]:
```suggestion
/// make it convenient to iterate through each property (see the fold below).
```

File: rust/xaynet-analytics/src/database/controller_data/adapter.rs [hunk 3]
```diff
@@ -8,6 +11,7 @@ use std::vec::IntoIter;
 
 use crate::database::common::{FieldProperty, IsarAdapter, SchemaGenerator};
 
+/// Allows to convert an IsarObject from the db to an `ControllerData`.
```
Comment (finiteprods) [inline rust/xaynet-analytics/src/database/controller_data/adapter.rs:14]:
```suggestion
/// Allows to convert an IsarObject from the db to a `ControllerData`.
```

File: rust/xaynet-analytics/src/database/controller_data/data_model.rs [hunk 4]
```diff
@@ -1,9 +1,13 @@
+//! In this file `ControllerData` is declared, together with tome conversion methods to/from adapters.
```
Comment (finiteprods) [inline rust/xaynet-analytics/src/database/controller_data/data_model.rs:1]:
```suggestion
//! In this file `ControllerData` is declared, together with some conversion methods to/from adapters.
```

File: rust/xaynet-analytics/src/database/screen_route/data_model.rs [hunk 5]
```diff
@@ -1,3 +1,5 @@
+//! In this file `ScreenRoute` is declared, together with tome conversion methods to/from adapters.
```
Comment (finiteprods) [inline rust/xaynet-analytics/src/database/screen_route/data_model.rs:1]:
```suggestion
//! In this file `ScreenRoute` is declared, together with some conversion methods to/from adapters.
```

File: rust/xaynet-analytics/src/database/screen_route/data_model.rs [hunk 6]
```diff
@@ -7,6 +9,7 @@ use crate::database::{
     screen_route::adapter::ScreenRouteAdapter,
 };
 
+/// A `ScreenRoute` is the internal representation of a screen in the app installing the library.
```
Comment (finiteprods) [inline rust/xaynet-analytics/src/database/screen_route/data_model.rs:12]: "of a screen in the app installing the library"?

**Review by finiteprods — APPROVED** [review]
just minor typos, otherwise good to go. thank you!

File: rust/xaynet-analytics/src/database/analytics_event/data_model.rs [hunk 7]
```diff
@@ -33,6 +42,13 @@ impl TryFrom<i32> for AnalyticsEventType {
     }
 }
 
+/// The core data model of the library. It represents an event recorded on the mobile framework side.
+/// It can be logged manually by the developers, or automatically detected by Flutter/the mobile framework side.
+/// ## Fields:
+/// * `name`: The name of the event.
+/// * `eventy_type`: The type of event.
```
[the remark this answers is not in the document]
Comment (wilk10) [reply rust/xaynet-analytics/src/database/analytics_event/data_model.rs:49]: again!? ahah

File: rust/xaynet-analytics/src/database/screen_route/data_model.rs (hunk 6 above)
Comment (wilk10) [reply rust/xaynet-analytics/src/database/screen_route/data_model.rs:12]: the screens of the app that will use federated analytics. is it clear? how should i change it?

Comment (finiteprods) [reply rust/xaynet-analytics/src/database/screen_route/data_model.rs:12]: ah ok, I was parsing it wrong. maybe just chop off "installing the library", should be clear enough.

**Pull request merged**